### PR TITLE
Enable pprof http server

### PR DESCRIPTION
**Pull request by cowsed**
# Description
Enabled the net/http/pprof server from the go standard library.

# How Has This Been Tested?
1. Run gsw_service with the `-p` flag: `go run ./cmd/gsw_service.go -p`
2. Download profiling data from the website: `go tool pprof http://localhost:12345/debug/pprof/heap`
3. Type `web`. Press enter. See cool graphs

# Checklist:
- [ ] New functionality is documented in the necessary spots (i.e new functions documented in the header)
- [ ] Unit tests cover any new functionality or edge cases that the PR was meant to resolve (if applicable) 
- [ ] The CI checks are passing
- [ ] I reviewed my own code in the GitHub diff and am sure that each change is intentional

### CODE REVIEW

File: cmd/gsw_service.go [hunk 1]
```diff
@@ -17,6 +17,9 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/AarC10/GSW-V2/proc"
+
+	"net/http"
+	_ "net/http/pprof"
```
Comment (AarC10) [inline cmd/gsw_service.go:22]: ?

File: cmd/gsw_service.go [hunk 2]
```diff
@@ -122,23 +126,30 @@ func readConfig() *viper.Viper {
...
-	config := readConfig()
+	config, doProfiling := readConfig()
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	if doProfiling {
+		go func() {
+			logger.Info("Running pprof server at localhost:12345")
+			http.ListenAndServe("localhost:12345", nil)
```
Comment (AarC10) [inline cmd/gsw_service.go:142]: Can the port be configurable?

**Review by AarC10 — CHANGES_REQUESTED** [review]

File: cmd/gsw_service.go (hunk 1 above)
Comment (cowsed) [reply cmd/gsw_service.go:22]: - net/http to start the server
- `_ "net/http/pprof" to bring in the pprof package. _ to keep it anonymous or else it would be an unused import. need to import it bc the init() in that package sets it up to listen on the global httpserver

File: cmd/gsw_service.go (hunk 2 above)
Comment (cowsed) [reply cmd/gsw_service.go:142]: yeah prolly

File: cmd/gsw_service.go [hunk 3]
```diff
@@ -122,23 +126,30 @@ func readConfig() *viper.Viper {
...
 	// Read gsw_service config
-	config := readConfig()
+	config, profilingPort := readConfig()
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	if profilingPort != 0 {
+		go func() {
+			logger.Info("Running pprof server at localhost:12345")
```
Comment (AarC10) [inline cmd/gsw_service.go:141]: Alright just fix this line lol

**Review by AarC10 — APPROVED** [review]

**Comment by AarC10** [comment]
You can also ignore the static analysis checks for now in this PR. Ill fix em in another one

**Comment by cowsed** [comment]
> You can also ignore the static analysis checks for now in this PR. Ill fix em in another one

i love ignoring static analysis

**Comment by apenza** [comment]
> You can also ignore the static analysis checks for now in this PR. Ill fix em in another one

This makes me sad because now we have unhandled errors but if Aaron says you can ignore them then everything else looks good!

**Review by apenza — APPROVED** [review]
